kolena/_experimental/object_detection/utils.py

```diff
@@ -16,12 +16,120 @@
 from typing import List
 from typing import Optional
 from typing import Set
+from typing import Tuple
 
+import numpy as np
+
+from kolena._extras.metrics.sklearn import sklearn_metrics
 from kolena._utils import log
 from kolena.workflow.evaluator import ConfusionMatrix
+from kolena.workflow.evaluator import Curve
+from kolena.workflow.evaluator import CurvePlot
 from kolena.workflow.metrics import MulticlassInferenceMatches
 
 
+def _compute_sklearn_arrays(
+    all_matches: List[MulticlassInferenceMatches],
+) -> Tuple[List[int], List[int], Dict[str, List[int]], Dict[str, List[int]]]:
+    y_true_by_label: defaultdict[str, List[int]] = defaultdict(lambda: [])
+    y_score_by_label: defaultdict[str, List[int]] = defaultdict(lambda: [])
+    y_true: List[int] = []
+    y_score: List[int] = []
+    for image_bbox_matches in all_matches:
+        for _, bbox_inf in image_bbox_matches.matched:  # TP (if above threshold)
+            y_true_by_label[bbox_inf.label].append(1)
+            y_score_by_label[bbox_inf.label].append(bbox_inf.score)
+            y_true.append(1)
+            y_score.append(bbox_inf.score)
+        for bbox_gt, _ in image_bbox_matches.unmatched_gt:  # FN
+            y_true.append(1)
+            y_score.append(-1)
+            y_true_by_label[bbox_gt.label].append(1)
+            y_score_by_label[bbox_gt.label].append(-1)
+        for bbox_inf in image_bbox_matches.unmatched_inf:  # FP (if above threshold)
+            y_true_by_label[bbox_inf.label].append(0)
+            y_score_by_label[bbox_inf.label].append(bbox_inf.score)
+            y_true.append(0)
+            y_score.append(bbox_inf.score)
+    return y_true, y_score, dict(y_true_by_label), dict(y_score_by_label)
+
+
+def _compute_threshold_curves(
+    y_true: np.ndarray,
+    y_score: np.ndarray,
+    label: str,
+) -> Tuple[Curve, Curve]:
+    if len(y_score) >= 501:
+        thresholds = list(np.linspace(min(abs(y_score)), max(y_score), 501))[:-1]
+    else:
+        thresholds = np.unique(np.sort(y_score))
+        thresholds = thresholds[thresholds >= 0]
+        thresholds = thresholds.tolist()[:-1]
+
+    precisions: List[float] = []
+    recalls: List[float] = []
+    f1s: List[float] = []
+    for threshold in thresholds:
+        y_pred = [1 if score > threshold else 0 for score in y_score]
+        precision, recall, f1, _ = sklearn_metrics.precision_recall_fscore_support(
+            y_true,
+            y_pred,
+            average="binary",
+            zero_division=0,
+        )
+
+        precisions.append(precision)
+        recalls.append(recall)
+        f1s.append(f1)
+    return Curve(x=thresholds, y=f1s, label=label), Curve(x=recalls, y=precisions, label=label)
```

Comment:
Minor, but feels like this interface should return lists, and the `Curve` should be created in the same context where the `CurvePlot` is created.